### PR TITLE
Remove JDK 8

### PR DESCRIPTION
**Pull request by gastaldi**
JDK 11 is now the minimum to build Quarkus

### CODE REVIEW

**Review by gsmet — CHANGES_REQUESTED** [review]
Ah, thanks! I had it on my list but even better if someone else does it :).

I think there are a few more to update such as all the Gradle projects. See here when I had to go the other way: https://github.com/quarkusio/quarkus/pull/16137/files (and all the other projects were targeting Java 8).

File: .github/workflows/release-build.yml
Comment (ivannov) [inline .github/workflows/release-build.yml:21]: Maybe a typo here? You meant 11 and not 1?

Comment (gastaldi) [reply .github/workflows/release-build.yml:21]: Ah good catch. Thanks!

Comment (gastaldi) [inline .github/workflows/release-build.yml:21]:
```suggestion
      - name: Set up JDK 11
```

File: pom.xml
Comment (famod) [inline pom.xml:53]: Are those properties really needed here? Or ist this more a "just in case" thing? I'm surely missing something here...

**Review by famod — CHANGES_REQUESTED** [review]
+💯 for kicking it out!

There are a few profile activations related to JDK 8, e.g. this in `build-parent`:
```xml
        <profile>
            <id>jdk-8-classpath</id>
            <activation>
                <!-- activate this on any JDK from 9 onwards -->
                <jdk>[9,)</jdk>
            </activation>
            <properties>
                <maven.compiler.release>8</maven.compiler.release>
            </properties>
        </profile>
 ```
 
I think `<jdk>` is a good search term to find all.

File: pom.xml
Comment (gastaldi) [reply pom.xml:53]: Yes, because jboss-parent sets JDK 8 as the source/target version

Comment (famod) [reply pom.xml:53]: Ok, but which module that is actually compiling something is _not_ an independent-project and does not inherit from `build-parent`? Are there any?

Comment (gastaldi) [reply pom.xml:53]: `build-parent` and `bom/application` and `bom/test` inherit from `quarkus-parent`. Looks like I should remove from build-parent

**Comment by gastaldi** [comment]
There is a bug in the `maven-compiler-plugin 3.8.1-jboss-1`: `independent-modules/arc` does not build when the jdk-8-classpath profile is removed. Any ideas @dmlloyd ?

**Comment by famod** [comment]
It just occured to me that once we merge this, backports might get more complicated as people will start using Java 11 APIs.

**Comment by gastaldi** [comment]
> It just occured to me that once we merge this, backports might get more complicated as people will start using Java 11 APIs.

This is planned for the 2.x releases. How would that be more complicated?

**Comment by famod** [comment]
I meant as long as 1.x branches shall receive backports from `main`, issues like that could pop up.

**Review by famod — CHANGES_REQUESTED** [review]
I think the JDK 8 profiles in those three pom.xml should be removed as well:
- `extensions/grpc/deployment/pom.xml`
- `integration-tests/grpc-mutual-auth/pom.xml`
- `integration-tests/grpc-tls/pom.xml`

It might then also be possible to remove `boring-ssl` from `build-parent`.

**Review by famod — APPROVED** [review]
LGTM now!

I've seen all those build failures before in other PRs so I don't think they are related.

**Comment by gastaldi** [comment]
@famod thank you! I'll let @gsmet have the final say and merge this

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
I thought about adding a thumbs up to the PR but you have 11, which is perfect :).

Thanks for this, let's merge and iterate from there if we find other things to adapt.

**Pull request merged**
